### PR TITLE
chore: fix compiler warnings

### PR DESCRIPTION
**Pull request by mjsax**
move off deprecated Apache Kafka APIs: KIP-633 and KIP-516
add suppress warnings where appropriate

### CODE REVIEW

File: ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/array/ArrayConcat.java [hunk 1]
```diff
@@ -40,7 +40,7 @@
     }
     final int leftSize = left != null ? left.size() : 0;
     final int rightSize = right != null ? right.size() : 0;
-    final List<T> result = new ArrayList(leftSize + rightSize);
+    final List<T> result = new ArrayList<>(leftSize + rightSize);
```
Comment (mjsax) [inline ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/array/ArrayConcat.java:43]: Side cleanup

File: ksqldb-engine/src/main/java/io/confluent/ksql/services/KafkaTopicClientImpl.java [hunk 2]
```diff
@@ -284,7 +284,7 @@ public void deleteTopics(final Collection<String> topicsToDelete) {
     }
 
     final DeleteTopicsResult deleteTopicsResult = adminClient.get().deleteTopics(topicsToDelete);
-    final Map<String, KafkaFuture<Void>> results = deleteTopicsResult.values();
+    final Map<String, KafkaFuture<Void>> results = deleteTopicsResult.topicNameValues();
```
Comment (mjsax) [inline ksqldb-engine/src/main/java/io/confluent/ksql/services/KafkaTopicClientImpl.java:287]: KIP-516

File: ksqldb-engine/src/test/java/io/confluent/ksql/services/KafkaTopicClientImplTest.java [hunk 3]
```diff
@@ -982,7 +987,7 @@ private void givenTopicConfigs(
           ));
 
       final DeleteTopicsResult deleteTopicsResult = mock(DeleteTopicsResult.class);
-      when(deleteTopicsResult.values()).thenReturn(result);
+      when(deleteTopicsResult.topicNameValues()).thenReturn(result);
```
Comment (mjsax) [inline ksqldb-engine/src/test/java/io/confluent/ksql/services/KafkaTopicClientImplTest.java:990]: KIP-516

File: ksqldb-engine/src/test/java/io/confluent/ksql/services/KafkaTopicClientImplTest.java [hunk 4]
```diff
@@ -1002,7 +1007,7 @@ private void givenTopicConfigs(
           );
 
       final DeleteTopicsResult deleteTopicsResult = mock(DeleteTopicsResult.class);
-      when(deleteTopicsResult.values()).thenReturn(result);
+      when(deleteTopicsResult.topicNameValues()).thenReturn(result);
```
Comment (mjsax) [inline ksqldb-engine/src/test/java/io/confluent/ksql/services/KafkaTopicClientImplTest.java:1010]: KIP-516

File: ksqldb-parser/src/main/java/io/confluent/ksql/parser/tree/WithinExpression.java [hunk 5]
```diff
@@ -108,12 +108,10 @@ public String toString() {
...
-    if (gracePeriod.isPresent()) {
-      builder.append(" GRACE PERIOD ")
-          .append(gracePeriod.get().getValue())
-          .append(' ')
-          .append(gracePeriod.get().getTimeUnit());
-    }
+    gracePeriod.ifPresent(windowTimeClause -> builder.append(" GRACE PERIOD ")
+        .append(windowTimeClause.getValue())
+        .append(' ')
+        .append(windowTimeClause.getTimeUnit()));
```
Comment (mjsax) [inline ksqldb-parser/src/main/java/io/confluent/ksql/parser/tree/WithinExpression.java:114]: Side cleanup

**Review by spena — APPROVED** [review]
LGTM

**Pull request merged**